### ISSUE TITLE
Error running devspace

### ISSUE THREAD

**Issue by joshughes** [covexo/devspace#233]
Just trying to start my first project... 

Just set up using all the default options. 

Kubectl works okay for my local K8 Cluster

Using Docker For Mac Kubernetes 

On MacOS using v1.0.3

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4d91ff3]

goroutine 1 [running]:
github.com/covexo/devspace/pkg/devspace/clients/kubectl.GetClientConfig(0x40, 0x555ea60, 0xc42130d201)
	/Users/travis/gopath/src/github.com/covexo/devspace/pkg/devspace/clients/kubectl/client.go:217 +0x53
github.com/covexo/devspace/pkg/devspace/clients/kubectl.NewClient(0xc420632500, 0x0, 0xc420723c00)
	/Users/travis/gopath/src/github.com/covexo/devspace/pkg/devspace/clients/kubectl/client.go:36 +0x22
github.com/covexo/devspace/cmd.(*UpCmd).Run(0xc42052d480, 0xc420885680, 0x68ea920, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/covexo/devspace/cmd/up.go:129 +0x7c
github.com/covexo/devspace/cmd.(*UpCmd).Run-fm(0xc420885680, 0x68ea920, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/covexo/devspace/cmd/up.go:95 +0x52
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).execute(0xc420885680, 0x68ea920, 0x0, 0x0, 0xc420885680, 0x68ea920)
	/Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x68af2a0, 0x0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).Execute(0x68af2a0, 0x0, 0x0)
	/Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/covexo/devspace/cmd.Execute()
	/Users/travis/gopath/src/github.com/covexo/devspace/cmd/root.go:39 +0x91
main.main()
	/Users/travis/gopath/src/github.com/covexo/devspace/main.go:15 +0x3e
```

**Comment by andrewrothman** [comment]
I'm getting the same issue as well on Docker for Mac with Kubernetes.

```
andrew [backend] devspace up                                                                                                                                                                                                         master
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4d92044]

goroutine 1 [running]:
github.com/covexo/devspace/pkg/devspace/clients/kubectl.GetClientConfig(0xc42086b2b0, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/pkg/devspace/clients/kubectl/client.go:218 +0xa4
github.com/covexo/devspace/pkg/devspace/clients/kubectl.NewClient(0xc42003e001, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/pkg/devspace/clients/kubectl/client.go:36 +0x22
github.com/covexo/devspace/cmd.(*UpCmd).Run(0xc4204eb1c0, 0xc420965900, 0x68ea920, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/cmd/up.go:129 +0x7c
github.com/covexo/devspace/cmd.(*UpCmd).Run-fm(0xc420965900, 0x68ea920, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/cmd/up.go:95 +0x52
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).execute(0xc420965900, 0x68ea920, 0x0, 0x0, 0xc420965900, 0x68ea920)
        /Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x68af2a0, 0x0, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/covexo/devspace/vendor/github.com/spf13/cobra.(*Command).Execute(0x68af2a0, 0x0, 0x0)
        /Users/travis/gopath/src/github.com/covexo/devspace/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/covexo/devspace/cmd.Execute()
        /Users/travis/gopath/src/github.com/covexo/devspace/cmd/root.go:39 +0x91
main.main()
        /Users/travis/gopath/src/github.com/covexo/devspace/main.go:15 +0x3e
```

**Comment by LukasGentele** [comment]
Thanks for reporting the issue. That should be easy to fix. I have a couple of questions to narrow the problem down:
1. Are you using Minikube?
2. If you look at PROJECT/.devspace/config.yaml: Is there cluster.useKubeConfig = true?
3. If you look at ~/.kube/config: Is there an entry in clusters[*].certificate-authority-data or clusters[*].certificate-authority? (please do not post the contents of the file publicly, as it may contain sensible secret keys)

**Comment by geekflyer** [comment]
Same error message here.
My answer to your questions above:
1. No
2. my config.yaml:
```yaml
➜  foobar cat ./.devspace/config.yaml    
cluster:
  useKubeConfig: true
devSpace:
  portForwarding:
  - labelSelector:
      release: myapp
    portMappings:
    - localPort: 80
      remotePort: 80
    resourceType: pod
  release:
    name: myapp
    namespace: myapp
  sync:
  - containerPath: /app
    labelSelector:
      release: myapp
    localSubPath: ./
    resourceType: pod
    uploadExcludePaths:
    - Dockerfile
    - .devspace/
    - chart/
    - node_modules/
images:
  default:
    build:
      engine:
        docker:
          enabled: true
    name: myapp
    registry: default
registries:
  default:
    url: gcr.io
services:
  tiller:
    release:
      namespace: myapp
version: v1
```
3. yes, lot's of ` certificate-authority-data` entries for the gke clusters I use
4. `/Users/christian`

Additional context: I'm using GKE and things like `kubectl get pods` just work fine with the current cluster I'm connected to.

**Comment by LukasGentele** [comment]
Yes, I think you are completely right. That was the problem. I did not ship the google cloud auth plugin for the kubectl go-client. I am already testing a fix. Thank you for all the details!

**Comment by LukasGentele** [comment]
@joshughes, @geekflyer, @andrewrothman: I just released v1.0.4 with several fixes for google kubernetes engine. You can upgrade your devspace version with: `devspace upgrade`
Let me know if this release works for you or if there are any other issues.
